### ISSUE TITLE
Error: spades-hammer, OS return value: 1 

### ISSUE THREAD

**Issue by osvatic** [ablab/spades#1313]
### Description of bug

Spades-hammer is terminating because of a segmentation fault while kmer-counting and reports an "OS return value: 1". 

Command line:
```
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades.py      -t      48      -m      1000    -k      21,31,41,51,61,71,81,91,101,111,121     --meta  --pe-12 1       /lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/data/interleave_DNA/JMF-2108-11-0010C.interleave.fastq.gz   -o      /lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/results/JMF-2108-11-0010C_spades 
```

Details from spades.log:
Analyses_Jay/JMF_2108_11/results/JMF-2108-11-0010C_spades/split_input/JMF-2108-11-0010C.interleave_1.fastq
  0:01:18.914   329M / 897M  INFO   K-mer Counting           (kmer_data.cpp             : 324)   Processed 1201470 reads
  0:01:18.915   329M / 897M  INFO   K-mer Counting           (kmer_data.cpp             : 315)   Processing /lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/results/JMF-2108-11-0010C_spades/split_input/JMF-2108-11-0010C.interleave_2.fastq
The program was terminated by segmentation fault
=== Stack Trace ===
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer(+0x1b3d7) [0x55bce55dd3d7]
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer(+0x1bf06) [0x55bce55ddf06]
/lib64/libc.so.6(+0x3e6f0) [0x7fbbc903e6f0]
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer(+0x8e83f) [0x55bce565083f]
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer(+0x908f3) [0x55bce56528f3]
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer(+0x4a1f6) [0x55bce560c1f6]
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer(+0x4a3d6) [0x55bce560c3d6]
/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/../lib/libgomp.so.1(+0x186d9) [0x7fbbc932f6d9]
/lib64/libc.so.6(+0x89c12) [0x7fbbc9089c12]
/lib64/libc.so.6(+0x10ecc0) [0x7fbbc910ecc0]

== Error ==  system call for: "['/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/.snakemake/conda/0ae2b4a3475f1b8ec4109c90dec7e0d2_/bin/spades-hammer', '/lisc/scratch/jmf/internal/Analyses_Jay/JMF_2108_11/results/JMF-2108-11-0010C_spades/corrected/configs/config.info']" finished abnormally, OS return value: 1
None

This seems to be specific to a certain dataset as I have no problems running SPAdes on others within this project. 

### spades.log

[spades.log](https://github.com/user-attachments/files/15785788/spades.log)

### params.txt

[params.txt](https://github.com/user-attachments/files/15785789/params.txt)

### SPAdes version

SPAdes 3.15.5

### Operating System

Linux-5.14.0-362.24.1.0.1.el9_3.x86_64-x86_64-with-glibc2.34

### Python Version

version: 3.12.3

### Method of SPAdes installation

unkown 

### No errors reported in spades.log

- [X] Yes

**Comment by osvatic** [comment]
I just noticed that this is also the subject of issue #1291.

**Comment by asl** [comment]
Does the problem reproduce with the latest SPAdes 4.0?

**Comment by osvatic** [comment]
Yes, this is reproducible in SPAdes 4.0. The OS return value is the same and it looks like it gets stuck at the same place.



[params.txt](https://github.com/user-attachments/files/15800712/params.txt)
[spades.log](https://github.com/user-attachments/files/15800713/spades.log)

**Comment by tynot** [comment]
The same error occurred. Does anyone know how to resolve this issue?

**Comment by osvatic** [comment]
I was not able to solve this issue. 

I have tried different versions of SPAdes (3.15, 4.0) and the issues persisted. I was using bbmap for q-score and adapter trimming but tried switching to fastp for the trimming. That seemed to help.
